cocotb/share/lib/embed/gpi_embed.c

```diff
@@ -229,53 +229,62 @@ int get_module_ref(const char *modname, PyObject **mod)
     return 0;
 }
 
+int parse_entry_point_str (
+    char *entry_point_str,
+    const char **entry_module_name,
+    const char **entry_func_name)
```

Comment:
Looking at [setuptools](https://github.com/pypa/setuptools/blob/682b6511ac67e021b542e74ce30e13fe52bc2da9/pkg_resources/__init__.py#L2484), the format there is `some.module:some.attr` - ie, the dots can appear after the colon too

Reply:
I am going to move this into Python. There I can much more easily express checking and handle issues, while more likely being correct. I'm with you on being able to add utility functions to Python. I will go through the __init__ refactor as a separate PR.